### ISSUE TITLE
Contribute upstream?

### ISSUE THREAD

**Issue by Jehan** [seagetch/gimp-painter#1]
Hello seagetch!

Years ago, you proposed a patch on GIMP, but it could never be integrated since it was not built on the trunk/master. Yet you clearly had everyone's interest: https://bugzilla.gnome.org/show_bug.cgi?id=522226

Now I see you are still working on it on a regular basis, and passed to GIMP 2.8. This is unfortunately still not our master branch so it makes it again very hard to integrate. Would you not be interested to work on your features upstream, as one of the GIMP core developers?
I'm pretty sure we would all be very interested to get your propositions on the main GIMP. It would take some time to adapt all your changes to the new code (especially since we switched many pieces of code to GEGL) undoubtedly, but I think it would be worth it.

This does not mean the death of GIMP painter. You can continue to work on this version. But ultimately, gathering everyone's efforts in a single awesome project would be worth it, no? Thanks!

**Comment by seagetch** [comment]
Hello Jehan,
Thanks for your proposal.
For GUI related matters, I noticed that policy between core development teams and me (especially multi window mode and toolbar ) are totally different, thus I forked  the branch.
I don't like core team's way of designing UI. I think other members outside project should give them some pressure to understand the users' real needs.

After seeing the positive feedback of users to the gimp-painter, project members understand the users' real needs, and many of features are ported to trunk, thus I think many other features are somehow integrated into master branch.
Remained features such as dst-in layer ops may be integrated if feel is used for layer operation.
I think it is better to keep gimp-painter as a separated project, to test new features in advance. Gimp project members can understand how to implement new features, and will implement  in project's coding manner.

Actually, I don't want to be annoyed anymore by some of project member's objection based on poor understanding of actual painting operation flow.
